samtranslator/model/sam_resources.py

```diff
@@ -867,8 +878,115 @@ def to_cloudformation(self, **kwargs):
             resources.append(role)
             synthetics_canary.ExecutionRoleArn = role.get_runtime_attr("arn")
 
+        if self.CanaryMetricAlarms:
+            self._validate_cloudwatch_alarms()
+            for alarm_dict in self.CanaryMetricAlarms:
+                resources.append(self._construct_cloudwatch_alarms(alarm_dict))
+
         return resources
 
+    def _validate_cloudwatch_alarms(self):
+        """Validates the CanaryMetricAlarms property in Serverless Canary
+
+        The property should follow the following structure
+
+        CanaryMetricAlarms:
+            - AlarmName:
+                MetricName (required): one of the metrics in VALID_CANARY_METRICS
+                Threshold (optional): any value of type double
+                ComparisonOperator (optional): any of the valid values (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cw-alarm.html#cfn-cloudwatch-alarms-comparisonoperator)
+                Period (optional): Integer that is 10, 30, 60, or any multiple of 60
+
+        Note: Alarm names are used as logical ids for their respective CloudWatchAlarm property so if user has multiple
+        alarms there should be no duplicate names as we don't want the alarms to override each other without the user's
+        knowledge
+
+        :raise: InvalidResourceException
+        """
+        # keeps list of alarm names to make sure there are no duplicates
+        list_of_alarm_names = []
+        for alarm_dict in self.CanaryMetricAlarms:
+
+            # Throw an error if there is more than one alarm in the array index, like for example
+            # CanaryMetricAlarms:
+            #     - Alarm1:
+            #         MetricName: SuccessPercent
+            #       Alarm2:
+            #         MetricName: SuccessPercent
+            #         Threshold: 90
+            #     - Alarm3:
+            #         MetricName: Failed
+            # throws an error for Alarm2 since its Alarm1 is already defined in that dict
+            if len(alarm_dict) != 1:
+                raise InvalidResourceException(self.logical_id, "Must only have one alarm per array index")
+
+            # get the alarm name and the properties the user defined for the alarm
+            alarm_name = list(alarm_dict.keys())[0]
```

Comment:
Also you are checking `len(alarm_dict) != 1`. What happens if size is zero? It'll probably through an index out of bounds (or something similar).

Reply:
There is a validation before this

Reply:
if the size is zero then the 
```
raise InvalidResourceException(self.logical_id, "Must have one alarm per array index")
``` 
will be triggered